staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go

```diff
@@ -62,6 +62,49 @@ func (r *BucketRateLimiter) NumRequeues(item interface{}) int {
 func (r *BucketRateLimiter) Forget(item interface{}) {
 }
 
+type ItemBucketRateLimiter struct {
+	r     rate.Limit
+	burst int
+
+	limitersLock sync.Mutex
+	limiters     map[interface{}]*rate.Limiter
+}
+
+var _ RateLimiter = &ItemBucketRateLimiter{}
+
+func NewItemBucketRateLimiter(r rate.Limit, burst int) *ItemBucketRateLimiter {
```

Comment:
Can we name r qps?
Also - can we simply use float32 instead of using rate.Limit?